api/v1alpha1/applicationset_types.go

```diff
@@ -112,6 +119,38 @@ type GitFileGeneratorItem struct {
 	Path string `json:"path"`
 }
 
+// RepoHostGenerator defines a generator that scrapes a SCMaaS API to find candidate repos.
+type RepoHostGenerator struct {
+	// Which provider to use and config for it.
+	Github *RepoHostGeneratorGithub `json:"github,omitempty"`
+	// TODO other providers.
+	// Filters for which repos should be considered.
+	Filters []RepoHostGeneratorFilter `json:"filters,omitempty"`
+	// Standard parameters.
+	RequeueAfterSeconds *int64                 `json:"requeueAfterSeconds,omitempty"`
+	Template            ApplicationSetTemplate `json:"template,omitempty"`
+}
+
+// RepoHostGeneratorGithub defines a connection info specific to Github.
+type RepoHostGeneratorGithub struct {
+	// Github org to scan. Required.
+	Organization string `json:"organization"`
+	// The Github API URL to talk to. If blank, use https://api.github.com/.
+	API string `json:"api,omitempty"`
+	// Authentication token reference.
+	TokenRef *SecretRef `json:"tokenRef,omitempty"`
+}
+
+// RepoHostGeneratorFilter is a single repoisitory filter.
```

Comment:
I think re: the first that `filters: []` and `filters: [{}]` will both technically work as no-ops since the filter code is fairly modular. The `[{}]` would be weird though and I would be fine if that was a validation error at some point.